tests/default_missing_vals.rs

```diff
@@ -0,0 +1,116 @@
+    assert!(r.is_ok());
+    let m = r.unwrap();
+    assert!(m.is_present("color"));
+    assert_eq!(m.value_of("color").unwrap(), "auto");
+}
+
+#[test]
+fn opt_present_with_no_value() {
+    let r = App::new("df")
+        .arg(Arg::new("color").long("color")
+            .value_name("WHEN")
+            .possible_values(&["always", "auto", "never"])
+            .multiple(true)
+            .overrides_with("color")
+            .about(
+                r#"Specify WHEN to colorize output.
+                [WHEN] == ( *always* | auto | never ) ; "--color=auto" is assumed if the option is never specified
+                "#,
+                )
+            .default_value("auto")
+            .hide_default_value(true)
+            .min_values(0)
+            .require_equals(true)
+            .default_missing_value("always")
+        )
+        .try_get_matches_from(vec!["", "--color"]);
+    assert!(r.is_ok());
+    let m = r.unwrap();
+    assert!(m.is_present("color"));
+    assert_eq!(m.value_of("color").unwrap(), "always");
+}
+
+#[test]
+fn opt_present_with_value() {
+    let r = App::new("df")
+        .arg(Arg::new("color")
+            .long("color")
+            .value_name("WHEN")
+            .possible_values(&["always", "auto", "never"])
+            .multiple(true)
+            .overrides_with("color")
+            .about(
+                r#"Specify WHEN to colorize output.
+                [WHEN] == ( *always* | auto | never ) ; "--color=auto" is assumed if the option is never specified
+                "#,
+                )
+            .default_value("auto")
+            .hide_default_value(true)
+            .min_values(0)
+            .require_equals(true)
+            .default_missing_value("always")
+        )
+        .try_get_matches_from(vec!["", "--color=never"]);
+    assert!(r.is_ok());
+    let m = r.unwrap();
+    assert!(m.is_present("color"));
+    assert_eq!(m.value_of("color").unwrap(), "never");
+}
+
+#[test]
+fn opt_default() {
+    // assert no change to usual argument handling when adding default_missing_value()
+    let r = App::new("df")
+        .arg(Arg::from("-o [opt] 'some opt'")
+            .default_value("default")
+            .default_missing_value("default_missing")
+        )
+        .try_get_matches_from(vec![""]);
+    assert!(r.is_ok());
+    let m = r.unwrap();
+    assert!(m.is_present("o"));
+    assert_eq!(m.value_of("o").unwrap(), "default");
+}
+
+#[test]
+fn opt_default_user_override() {
+    // assert no change to usual argument handling when adding default_missing_value()
+    let r = App::new("df")
+        .arg(Arg::from("--opt [FILE] 'some arg'")
+            .default_value("default")
+            .default_missing_value("default_missing")
+        )
+        .try_get_matches_from(vec!["", "--opt", "value"]);
+    assert!(r.is_ok());
+    let m = r.unwrap();
+    assert!(m.is_present("opt"));
+    assert_eq!(m.value_of("opt").unwrap(), "value");
+}
+
+}
```

Comment:
?

Reply:
I'm not sure what the question mark is for ... I added essentially duplicates of a couple of the "default_value" tests to make sure there wasn't any outcome change when the "default_missing_value" was added. They are just a quick sanity check. I didn't think them completely through... I can drop or change them.

If it's the last `}`... I needed to be able to see just these tests, so I added them into a `default_missing_vals` module (so I could `cargo test default_missing`).